src/uct/ib/rc/base/rc_iface.c

```diff
@@ -877,12 +901,27 @@ ucs_status_t uct_rc_iface_event_arm(uct_iface_h tl_iface, unsigned events)
 ucs_status_t uct_rc_iface_fence(uct_iface_h tl_iface, unsigned flags)
 {
     uct_rc_iface_t *iface = ucs_derived_of(tl_iface, uct_rc_iface_t);
+    ucs_status_t status;
+    uct_rc_ep_t *ep;
 
-    if (iface->config.fence_mode != UCT_RC_FENCE_MODE_NONE) {
-        iface->tx.fi.fence_beat++;
+    ucs_list_for_each(ep, &iface->ep_list, list) {
+        status = uct_ep_fence(&ep->super.super, flags);
+        if (status != UCS_OK) {
```

Comment:
I'd  `assert(status==UCS_OK)` instead, because it should always succeed

Reply:
nop - it may return out-of-memory